frontend/src/components/Listing/ListingGrid.tsx

```diff
@@ -1,79 +1,136 @@
-                    defaultValue={"Alphabetic"}
-                    inputProps={{
-                        name: 'listingSorting',
-                        id: 'sortListings',
-                    }}
-                >
-                    <MenuItem value={"Alphabetic"} onClick={() => setSortingType("Alphabetic")}>Alphabetic (A-Z)</MenuItem>
-                    <MenuItem value={"ReverseAlphabetic"} onClick={() => setSortingType("ReverseAlphabetic")}>Reverse Alphabetic (Z-A)</MenuItem>
-                    <MenuItem value={"Expensive"} onClick={() => setSortingType("Expensive")}>Price (Most expensive)</MenuItem>
-                    <MenuItem value={"Cheapest"} onClick={() => setSortingType("Cheapest")}>Price (Cheapest)</MenuItem>
-                    <MenuItem value={"AuthorAlphabetic"}onClick={() => setSortingType("AuthorAlphabetic")}>Author Alphabetic</MenuItem>
-                </Select>
-            </FormControl>
-          <Grid container spacing={{ xs: 1, sm: 4 }} columns={{ xs: 1, sm: 8, md: 12 }}>
+      )}
+      {listings.length !== 0 && (
+        <Container maxWidth="xl" className="grid">
+          <FormControl>
+            <InputLabel variant="standard" htmlFor="uncontrolled-native">
+              Listing Sorting
+            </InputLabel>
+            <Select
+              defaultValue={"Alphabetic"}
+              inputProps={{
+                name: "listingSorting",
+                id: "sortListings",
+              }}
+            >
+              <MenuItem
+                value={"Alphabetic"}
+                onClick={() => setSortingType("Alphabetic")}
+              >
+                Alphabetic (A-Z)
+              </MenuItem>
+              <MenuItem
+                value={"ReverseAlphabetic"}
+                onClick={() => setSortingType("ReverseAlphabetic")}
+              >
+                Reverse Alphabetic (Z-A)
+              </MenuItem>
+              <MenuItem
+                value={"Expensive"}
+                onClick={() => setSortingType("Expensive")}
+              >
+                Price (Most expensive)
+              </MenuItem>
+              <MenuItem
+                value={"Cheapest"}
+                onClick={() => setSortingType("Cheapest")}
+              >
+                Price (Cheapest)
+              </MenuItem>
+              <MenuItem
+                value={"AuthorAlphabetic"}
+                onClick={() => setSortingType("AuthorAlphabetic")}
+              >
+                Author Alphabetic
+              </MenuItem>
+              <MenuItem
+                value={"ReleaseDateOldest"}
+                onClick={() => setSortingType("ReleaseDateOldest")}
+              >
+                Release Date (Oldest to Newest)
+              </MenuItem>
+              <MenuItem
+                value={"ReleaseDateNewest"}
+                onClick={() => setSortingType("ReleaseDateNewest")}
+              >
+                Release Date (Newest to Oldest)
+              </MenuItem>
```

Comment:
This can be refactored into an interation over a list to help with readability and reducing code reuse

Define this const before the return value of this component
```
  const sortMenuOptions = [
    {
      sortingType: "Alphabetic",
      text: "Alphabetic (A-Z)",
    },
    {
      sortingType: "ReverseAlphabetic",
      text: "Reverse Alphabetic (Z-A)",
    },
    {
      sortingType: "Expensive",
      text: "Price (Most expensive)",
    },
    {
      sortingType: "Cheapest",
      text: "Price (Cheapest)",
    },
    {
      sortingType: "AuthorAlphabetic",
      text: "Author Alphabetic",
    },
    {
      sortingType: "ReleaseDateOldest",
      text: "Release Date (Oldest to Newest)",
    },
    {
      sortingType: "ReleaseDateNewest",
      text: "Release Date (Newest to Oldest)",
    },
  ];
```

The replace all of the currently hardcoded MenuItems with the below block
```
{sortMenuOptions.map(menuItem => {
                return (
                  <MenuItem
                    value={menuItem.sortingType}
                    onClick={() => setSortingType(menuItem.sortingType)}
                  >
                    {menuItem.text}
                  </MenuItem>
                );
              })}
```

Reply:
Refactored